src/v/raft/replicate_batcher.cc

```diff
@@ -31,15 +31,11 @@ replicate_stages replicate_batcher::replicate(
   consistency_level consistency_lvl) {
     ss::promise<> enqueued;
     auto enqueued_f = enqueued.get_future();
-    try {
+
         auto f = cache_and_wait_for_result(
           std::move(enqueued), expected_term, std::move(r), consistency_lvl);
         return {std::move(enqueued_f), std::move(f)};
-    } catch (...) {
-        return {
-          ss::current_exception_as_future<>(),
-          ss::make_ready_future<result<replicate_result>>(errc::shutting_down)};
-    }
```

Comment:
looks like the previous comment is no longer relevant since the outer try block is removed.

but wasn't this outer try block handling this case:

```
          if (f.failed()) {
              enqueued.set_exception(f.get_exception());
              return ss::make_ready_future<result<replicate_result>>(
                errc::replicate_batcher_cache_error);
          }
```

from the non-coroutine version when flush or lock/unit acquisition might fail?

Reply:
indeed it was, but currently both of the exceptions will be handled in `handle_exception` continuation of back grounded flush. I will maybe change the commit ordering to make it clear that it is not need after i dispatch `flush` in background ?